lib/metabuf.js

```diff
@@ -0,0 +1,189 @@
+'use strict';
+
+const crypto = require('crypto');
+
+// starting index, number of bytes
+const fieldSpecs = {
+  1: [0, 1],
+  2: [1, 20],
+  4: [21, 8],
+  8: [29, 1],
+  24: [29, 1],      // flags | sample
+}
+
+class Metabuf {
+  constructor (metabuf) {
+    this.buf = Buffer.allocUnsafe(30);
+    this.buf[0] = 0x2b;
+    if (metabuf instanceof Metabuf) {
+      // a prototype was provided so use its taskId and flags but create
+      // a new, random opId for it.
+      metabuf.buf.copy(this.buf, 1, 1, 21);
+      this.buf[29] = metabuf.buf[29];
+      crypto.randomFillSync(this.buf, 21, 8);
+    } else if (arguments.length === 0) {
+      this.buf[29] = 0x00;
```

Comment:
it's ok.

Reply:
i guess this is my snarkiest answer but i'm curious - what is the concern? yes, it could expose 28 bytes but any bad node code could just allocate their own unfilled buffers to get the same effect.

Reply:
I guess in Ruby I look a the xtrace string in testing and often I will test for 2B000000...
I don't know if I have situations outside of tests, where I would like a correctly filled string.
I think in node there are checks for ('00000000',42) something, but maybe that was before refactoring.

Reply:
yes, looks like those checks are gone